model/github_app_auth_test.go

```diff
@@ -0,0 +1,61 @@
+package model
+
+import (
+	"testing"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindOneGithubAppAuth(t *testing.T) {
+	assert := assert.New(t)
+
+	require.NoError(t, db.Clear(GitHubAppAuthCollection),
+		"Error clearing collection")
+
+	key := []byte("I'm private!")
+	githubAppAuth := GithubAppAuth{
+		Id:         "mongodb",
+		AppId:      1234,
+		PrivateKey: key,
+	}
+	change, err := githubAppAuth.Upsert()
+	assert.NotNil(change)
+	assert.NoError(err)
+	assert.NotNil(change.UpsertedId)
+	assert.Equal(1, change.Updated, "%+v", change)
+
+	githubAppAuthFromDB, err := FindOneGithubAppAuth("mongodb")
+	assert.NoError(err)
+
+	assert.Equal("mongodb", githubAppAuthFromDB.Id)
+	assert.Equal(int64(1234), githubAppAuthFromDB.AppId)
+	assert.Equal(key, githubAppAuthFromDB.PrivateKey)
+}
+
+func TestRemove(t *testing.T) {
```

Comment:
If/when we rename the Remove function, we should also rename this to improve signaling on which feature failed!